src/details/ArborX_DetailsUtils.hpp

```diff
@@ -61,8 +62,10 @@ class ExclusiveScanFunctor
  *
  *  \pre \p src and \p dst must be of rank 1 and have the same size.
  */
-template <typename ST, typename... SP, typename DT, typename... DP>
-void exclusivePrefixSum(Kokkos::View<ST, SP...> const &src,
+template <typename ExecutionSpace, typename ST, typename... SP, typename DT,
+          typename... DP>
+void exclusivePrefixSum(ExecutionSpace &&space,
```

Comment:
Why is it being passed through move? The previous PR was using `ExecutionSpace const&`.

Reply:
It is a forwarding reference.

Admittedly it makes things a bit more complicated because if you pass a lvalue, `ExecutionSpace` is deduced to be a reference type.

Reply:
(That's what the parallel algorithms of the standard library do)

Reply:
It's even a universal reference so it binds to anything aggressively.